### PR TITLE
WIP Add macos as a testable platform to Travis CI

### PR DESCRIPTION
**Pull request by jscook2345**
**Description of the Pull Request (PR):**

Add macos as a testable platform to Travis CI

**Before submitting a PR, make sure you have done the following:**

[ ] Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
[ ] Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
[ ] Added tests to validate this PR and tested this PR locally with a `make testall`
[ ] Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
[ ] Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Comment by jscook2345** [comment]
This commit breaks suse RPMs.

For some reason the process is installing the files into `libexec` instead of `lib` and they are not found when building the RPM.

<pre>
Processing files: singularity-3.0.2-399.g9a80e4c9.x86_64
error: File not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity/bin/starter-suid
error: Directory not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity
error: File not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity/bin/starter
error: File not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity/cni/*

RPM build errors:
    File not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity/bin/starter-suid
    Directory not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity
    File not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity/bin/starter
    File not found: /home/vagrant/rpmbuild/BUILDROOT/singularity-3.0.2-399.g9a80e4c9.x86_64/usr/lib/singularity/cni/*
make: *** [Makefile:479: rpm] Error 1
</pre>

File: .travis/macos/test [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# man 1 bash
+set -o errexit # set -e
+set -o nounset # set -u
+set -o pipefail
+set -x
+
+sw_vers
+command -v singularity
+singularity version
+singularity help
```
Comment (ArangoGutierrez) [inline .travis/macos/test:11]: I would add a small test for the supported functionalities on macOS, like doing a pull, and a basic `if -f` (check for file existence bc we can not run it)

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Looks good for me.

File: .travis/macos/test (hunk 1 above)
Comment (jscook2345) [reply .travis/macos/test:11]: Done as per discussion. Did a search / pull with file check. More tests will be built into the platform at a later date.

**Review by ArangoGutierrez — APPROVED** [review]

**Review by jmstover — APPROVED** [review]

**Comment by ArangoGutierrez** [comment]
I can not update branch via GitHub UI @jscook2345 PING

**Pull request merged**